onset_maproom/calc.py

```diff
@@ -164,22 +289,51 @@ def strftimeb2int(strftimeb):
     strftimebint = strftimeb_all[strftimeb]
     return strftimebint
 
-
 def daily_tobegroupedby_season(
     daily_data, start_day, start_month, end_day, end_month, time_coord="T"
 ):
-    """Returns dataset ready to be grouped by with:
-    the daily data where all days not in season of interest are dropped
-    season_starts:
-      an array where the non-dropped days are indexed by the first day of their season
-      -- to use to groupby
-    seasons_ends: an array with the dates of the end of the seasons
-    Can then apply groupby on daily_data against seasons_starts,
-    and preserving seasons_ends for the record
-    If starting day-month is 29-Feb, uses 1-Mar.
-    If ending day-month is 29-Feb, uses 1-Mar and uses < rather than <=
-    That means that the last day included in the season will be 29-Feb in leap years
-    and 28-Feb otherwise
+    """ Grouping daily data by season.
+    
+        Returns dataset ready to be grouped by with the daily data where all days not in season of interest are dropped.
+
+            season_starts:
+                An array where the non-dropped days are indexed by the first day of their season.
+                -- to use to groupby
+            seasons_ends: an array with the dates of the end of the seasons.
+                Can then apply groupby on daily_data against seasons_starts, and preserving seasons_ends for the record.
+    
+        If starting day-month is 29-Feb, uses 1-Mar.
+    
+        If ending day-month is 29-Feb, uses 1-Mar and uses < rather than <=
+            That means that the last day included in the season will be 29-Feb in leap years and 28-Feb otherwise.
+
+        Parameters
+        -----------
+        daily_data : DataArray
+                    Daily data to be grouped.
+        start_day : int
+                    Day of the start date  of the season.
+        start_month : int
+                    Month of the start date of the season.
+        end_day : int
+                    Day of the end date of the season.
+        end_month : int
+                    Day of the end date of the season.
+        time_coord : str, optional
+                    Time grid in 'daily_data' (the default is "T", which implies
+                    naming convention of time grid as "T" in 'soil_moisture').
+        Returns
+        -------
+        daily_tobegroupedby_season : Dataset
+                    Daily data grouped by season using season start date. Dataset includes grouped data
+                    and 'season_starts,' 'season_ends' as output variables.  
```

Comment:
actually the function is not grouping. It is "ready" to have groupby applied to it. But I can rephrase that later if you prefer.

Reply:
I always have to return to the source code of this function to figure out what it does, because the name doesn't remind me. Can we rename it something like `add_seasons`, `label_seasons`, `extract_seasons`, ...?

(Possibly a moot point eventually, if we switch to the approach of making season year a separate dimension.)

Reply:
I am happy to do that in another PR.

Reply:
Yes that sounds like something to do in a new PR, but would be helpful.

Reply:
I made this an issue (not that I am really using the Issue tracker much as of yet but need to get started somewhere...). Thus, I resolve.